packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridRow.js

```diff
@@ -18,12 +19,39 @@ const blockClass = `${pkg.prefix}--datagrid`;
 
 const { TableRow, TableCell } = DataTable;
 
+const rowHeights = {
+  xs: 24,
+  sm: 32,
+  md: 40,
+  lg: 48,
+  xl: 64,
+};
+
 // eslint-disable-next-line react/prop-types
 const DatagridRow = (datagridState) => {
-  const { row } = datagridState;
+  const { row, rowSize, withNestedRows } = datagridState;
   const { state } = useContext(InlineEditContext);
   const { activeCellId } = state;
   const activeCellObject = activeCellId && getCellIdAsObject(activeCellId);
+
+  const getVisibleNestedRowCount = ({ isExpanded, subRows }) => {
```

Comment:
i think i get the gist of what this is trying to do. i think you can clean this up by using recursion instead of nesting if and foreach statements. you could do something like this
```
const getRowHeight = ({ isExpanded, subRows }) => {
  let size = 0;
  if (isExpanded && subRows) {
    size += subRows.length;
    subRows.forEach((child) => {
      size += getRowHeight(child);
    })
  };
  return size;
};
```

Reply:
Ah yes, I like this much better too! I'll update it